[BUG]: Pytorch tests failing

### Contact Details [Optional]

johko@posteo.de

### System Information

{
  "cfg": {
    "data_backend": "mongodb://localhost:27017",
    "vector_search": "inmemory://",
    "artifact_store": null,
    "metadata_store": null,
    "cluster": {
      "distributed": false,
      "deserializers": [],
      "serializers": [],
      "dask_scheduler": "tcp://localhost:8786",
      "local": true,
      "backfill_batch_size": 100
    },
    "apis": {
      "retry": {
        "stop_after_attempt": 2,
        "wait_max": 10.0,
        "wait_min": 4.0,
        "wait_multiplier": 1.0
      }
    },
    "logging": {
      "level": "INFO",
      "type": "STDERR",
      "kwargs": {}
    },
    "server": {
      "host": "127.0.0.1",
      "port": 3223,
      "protocol": "http"
    },
    "downloads": {
      "hybrid": false,
      "root": "data/downloads"
    }
  },
  "cwd": "/home/johannes/Projects/superduperdb",
  "git": {
    "branch": "('branch', '--show-current') failed with [Errno 2] No such file or directory: 'branch'",
    "commit": "('show', '-s', '--format=\"%h: %s\"') failed with [Errno 2] No such file or directory: 'show'"
  },
  "hostname": "johannes-kolbe-hh-celebrate",
  "os_uname": [
    "Linux",
    "johannes-kolbe-hh-celebrate",
    "5.15.0-84-generic",
    "#93-Ubuntu SMP Tue Sep 5 17:16:10 UTC 2023",
    "x86_64"
  ],
  "package_versions": {},
  "platform": {
    "platform": "Linux-5.15.0-84-generic-x86_64-with-glibc2.35",
    "python_version": "3.8.18"
  },
  "startup_time": "2023-09-25 10:45:44.305842",
  "superduper_db_root": "/home/johannes/Projects/superduperdb",
  "sys": {
    "argv": [
      "/home/johannes/Projects/superduperdb/superduperdb/__main__.py",
      "info"
    ],
    "path": [
      "/home/johannes/Projects/superduperdb",
      "/usr/lib/python38.zip",
      "/usr/lib/python3.8",
      "/usr/lib/python3.8/lib-dynload",
      "/home/johannes/Projects/superduperdb/.venv/lib/python3.8/site-packages",
      "__editable__.superduperdb-0.0.7.finder.__path_hook__"
    ]
  }
}

### What happened?

When running `make test` a couple of tests in test_torch_utils.py are failing, related to pytorch device assignment (see log output below)

I'm not completely sure if that is a thing that is only failing for me, but think it is rather general.
I also have a branch ready that fixes these tests: https://github.com/johko/superduperdb/tree/fix_torch_device_tests and can create a PR if you agree

### Steps to reproduce

1. run `make test` from the terminal


### Relevant log output

```shell
FAILED test/unittest/model/test_torch_utils.py::test_device_of_cuda - AssertionError: assert device(type='cuda', index=0) == 'cuda'
FAILED test/unittest/model/test_torch_utils.py::test_set_device_context_manager - AssertionError: assert device(type='cuda', index=0) == 'cuda'
FAILED test/unittest/model/test_torch_utils.py::test_to_device_tensor - AssertionError: assert device(type='cuda', index=0) == device(type='cuda')
FAILED test/unittest/model/test_torch_utils.py::test_to_device_nested_list - AssertionError: assert device(type='cuda', index=0) == device(type='cuda')
FAILED test/unittest/model/test_torch_utils.py::test_to_device_nested_dict - AssertionError: assert device(type='cuda', index=0) == device(type='cuda')
```

--- a/test/unittest/model/test_torch_utils.py
+++ b/test/unittest/model/test_torch_utils.py
@@ -24,7 +24,7 @@ def test_device_of_cuda(model):
     if torch.cuda.is_available():
         model.to(torch.device('cuda'))
         device = device_of(model)
-        assert device == 'cuda'
+        assert device.type == 'cuda'
 
 
 @pytest.mark.skipif(not torch, reason='Torch not installed')
@@ -37,16 +37,18 @@ def test_eval_context_manager(model):
 def test_set_device_context_manager(model):
     device_before = device_of(model)
     if torch.cuda.is_available():
-        with set_device(model, torch.device('cuda')):
+        with set_device(model, torch.device('cuda', 0)):
             device_after = device_of(model)
-            assert device_after == 'cuda'
+            assert device_after.type == 'cuda'
         assert device_of(model) == device_before
 
 
 @pytest.mark.skipif(not torch, reason='Torch not installed')
 def test_to_device_tensor(model):
     tensor = torch.tensor([1, 2, 3])
-    device = torch.device('cuda') if torch.cuda.is_available() else torch.device('cpu')
+    device = (
+        torch.device('cuda', 0) if torch.cuda.is_available() else torch.device('cpu')
+    )
     tensor_device = to_device(tensor, device)
     assert tensor_device.device == device
 
@@ -57,7 +59,9 @@ def test_to_device_nested_list(model):
         torch.tensor([1, 2, 3]),
         [torch.tensor([4, 5]), torch.tensor([6, 7])],
     ]
-    device = torch.device('cuda') if torch.cuda.is_available() else torch.device('cpu')
+    device = (
+        torch.device('cuda', 0) if torch.cuda.is_available() else torch.device('cpu')
+    )
     nested_list_device = to_device(nested_list, device)
     for item in nested_list_device:
         if isinstance(item, list):
@@ -69,7 +73,9 @@ def test_to_device_nested_list(model):
 @pytest.mark.skipif(not torch, reason='Torch not installed')
 def test_to_device_nested_dict(model):
     nested_dict = {'a': torch.tensor([1, 2, 3]), 'b': {'c': torch.tensor([4, 5])}}
-    device = torch.device('cuda') if torch.cuda.is_available() else torch.device('cpu')
+    device = (
+        torch.device('cuda', 0) if torch.cuda.is_available() else torch.device('cpu')
+    )
     nested_dict_device = to_device(nested_dict, device)
     for item in nested_dict_device.values():
         if isinstance(item, dict):

